wasm/datatypes/configuration.py

```diff
@@ -0,0 +1,62 @@
+from typing import (
+    NamedTuple,
+)
+
+from wasm.typing import (
+    Store,
+)
+
+from .instructions import (
+    InstructionSequence,
+)
+from .stack import (
+    Frame,
+    FrameStack,
+    Label,
+    LabelStack,
+    ValueStack,
+)
+
+
+class Configuration(NamedTuple):
+    store: Store
+    value_stack: ValueStack
+    label_stack: LabelStack
+    frame_stack: FrameStack
+
+    @property
+    def frame(self) -> Frame:
+        return self.frame_stack.peek()
+
+    @property
+    def has_active_frame(self):
+        return bool(self.frame_stack)
+
+    @property
+    def label(self) -> Label:
+        if self.has_active_frame and self.has_active_label:
+            return self.label_stack.peek()
+        else:
+            raise IndexError("No labels active for current frame")
+
+    @property
+    def has_active_label(self):
+        if not self.has_active_frame:
+            return False
+        elif not self.label_stack:
+            return False
+        else:
+            return self.label_stack.peek().frame_id == self.frame.id
+
+    @property
+    def instructions(self) -> InstructionSequence:
```

Comment:
I'm not confident in this API but it's working right now.  My spider sense says:

- These `property` methods are expensive
- The frame/label juggling may be error prone or it may only work incidentally but not actually be truly spec compliant.

Reply:
>These `property` methods are expensive

You mean compared to simple methods? According to my `timeit` a measurable performance different comes only into play for the *instance creation*  of `Configuration` because that is where the `@property` decorator syntax is evaluated. Once, the instance is created, there is no measurable difference between calling a *property*  or a *method*.

In other words, considering the performance implication of the `@property` syntax should only matter if we happen to create lots of `Configuration` instances repeatedly (which I doubt is happening, is it?)

Reply:
I'll profile this but the performance hit comes from the inner logic of checking which frame or label it should get the instruction sequence from.   In the previous implementation this was just a mutable property that was continually overwritten with whatever the active instruction sequence was.